lib/simctl.js

```diff
@@ -75,7 +75,11 @@ async function createDevice (name:string, deviceTypeId:string,
 }
 
 async function deleteDevice (udid:string):void {
-  await simExec('delete', 0, [udid]);
+  let loopFn:Function = async () => {
+    await simExec('delete', 0, [udid]);
+  };
+  // retry delete with a sleep in between because it's flakey
+  await retryInterval(5, 200, loopFn);
```

Comment:
Or we can catch stderr and emit nice errors. Then you can catch those errors on your side. etc.